pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -72,6 +76,25 @@ func (r *RedisReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	// 3scale 2.14 -> 2.15 Upgrade
+	// delete NAMESPACE attribute from secret system-redis
+	// and delete REDIS_NAMESPACE env var from deployments system-app and system-sidekiq
+	res, err := r.deleteSystemRedisSecretNamespaceAttribute()
```

Comment:
@valerymo do you know if this happens before or after migrating to deployments?

Reply:
happens before the migration, but when deployment is already up. Which means that the deployment (line 90) won't have the env anymore anyway due to the other changes in this PR.
Meaning that the deleteDeployment envar is not necessary but I think best would be to coordinate with @carlkyrillos on this

Reply:
Alternative is, if @carlkyrillos won't filter out the env, is that you could change your removal of env to target deploymentConfiguration instead, since at this point. The dcs are up, the deployment is up, the migration has not started. 
If you remove env from the dc and Carl migrates all envs, it should be ok. But IMO, it's an unnecessary step

Reply:
@MStokluska , we talked with you, when you helped to find solution how to implement it, and seems we wanted to place Before. But following your question --  seems to me - to update Secret (data)- it maybe doesn't matter (?), but for deployments env vars update - or after Migration , when we already have ddeployments (maybe need wait ...) or in time of Migration.  Thanks for comment

Reply:
As agreed, please move the secret key removal to the post migration step